### PR TITLE
bugfix: add support for Bronco Sport 2023

### PR DESCRIPTION
**Pull request by MohaElder**
**Car**
Ford 2023 Bronco Sport Copilot360+

**Route**
A route with the fingerprint
selfdrive/car/ford/fingerprints.py

**Reference carFw**
```
'carFw': [{'address': 2000,
                          'brand': 'ford',
                          'bus': 1,
                          'ecu': 'debug',
                          'fwVersion': b'1U5T-14G374-EM\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
                          'logging': True,
                          'obdMultiplexing': True,
                          'request': [b'>\x00', b'"\xf1\x88'],
                          'responseAddress': 2008,
                          'subAddress': 0},
                         {'address': 2016,
                          'brand': 'ford',
                          'bus': 1,
                          'ecu': 'engine',
                          'fwVersion': b'P1PA-14C204-AE\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
                          'logging': True,
                          'obdMultiplexing': True,
                          'request': [b'>\x00', b'"\xf1\x88'],
                          'responseAddress': 2024,
                          'subAddress': 0},
                         {'address': 1842,
                          'brand': 'ford',
                          'bus': 0,
                          'ecu': 'shiftByWire',
                          'fwVersion': b'PZ1P-14G395-AC\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
                          'logging': True,
                          'obdMultiplexing': True,
                          'request': [b'>\x00', b'"\xf1\x88'],
                          'responseAddress': 1850,
                          'subAddress': 0},
                         {'address': 1840,
                          'brand': 'ford',
                          'bus': 0,
                          'ecu': 'eps',
                          'fwVersion': b'LX6C-14D003-AL\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
                          'logging': False,
                          'obdMultiplexing': True,
                          'request': [b'>\x00', b'"\xf1\x88'],
                          'responseAddress': 1848,
                          'subAddress': 0},
                         {'address': 1888,
                          'brand': 'ford',
                          'bus': 0,
                          'ecu': 'abs',
                          'fwVersion': b'LX6C-2D053-RF\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
                          'logging': False,
                          'obdMultiplexing': True,
                          'request': [b'>\x00', b'"\xf1\x88'],
                          'responseAddress': 1896,
                          'subAddress': 0},
                         {'address': 1798,
                          'brand': 'ford',
                          'bus': 0,
                          'ecu': 'fwdCamera',
                          'fwVersion': b'M1PT-14F397-AD\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
                          'logging': False,
                          'obdMultiplexing': True,
                          'request': [b'>\x00', b'"\xf1\x88'],
                          'responseAddress': 1806,
                          'subAddress': 0},
                         {'address': 1892,
                          'brand': 'ford',
                          'bus': 0,
                          'ecu': 'fwdRadar',
                          'fwVersion': b'LB5T-14D049-AB\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
                          'logging': False,
                          'obdMultiplexing': True,
                          'request': [b'>\x00', b'"\xf1\x88'],
                          'responseAddress': 1900,
                          'subAddress': 0},
                         {'address': 1840,
                          'brand': 'ford',
                          'bus': 0,
                          'ecu': 'eps',
                          'fwVersion': b'\xff\x00\x00\x00\x02\x00\x02S\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
                          'logging': True,
                          'obdMultiplexing': True,
                          'request': [b'>\x00', b'"\xde\x00'],
                          'responseAddress': 1848,
                          'subAddress': 0},
                         {'address': 1798,
                          'brand': 'ford',
                          'bus': 0,
                          'ecu': 'fwdCamera',
                          'fwVersion': b'\x02;\x99\xdb\xa9 ze:\xa4U\xd0\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
                          'logging': True,
                          'obdMultiplexing': True,
                          'request': [b'>\x00', b'"\xde\x00'],
                          'responseAddress': 1806,
                          'subAddress': 0},
                         {'address': 1840,
                          'brand': 'ford',
                          'bus': 0,
                          'ecu': 'eps',
                          'fwVersion': b'\xff\xff\xff\xff\xff\xff\xff\xff\xff\x00\xff\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
                          'logging': True,
                          'obdMultiplexing': True,
                          'request': [b'>\x00', b'"\xde\x01'],
                          'responseAddress': 1848,
                          'subAddress': 0},
                         {'address': 1888,
                          'brand': 'ford',
                          'bus': 0,
                          'ecu': 'abs',
                          'fwVersion': b'\x01\x01\x05\x07\x01\x02\x02\x02\x04\x02\x01\x00\x08\xfe\x01\x03\x01\x00\x00\x00\x00',
                          'logging': True,
                          'obdMultiplexing': True,
                          'request': [b'>\x00', b'"\xde\x01'],
                          'responseAddress': 1896,
                          'subAddress': 0},
                         {'address': 1840,
                          'brand': 'ford',
                          'bus': 0,
                          'ecu': 'eps',
                          'fwVersion': b'\xff\x00\xff\x01\x01\x00\x02\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
                          'logging': True,
                          'obdMultiplexing': True,
                          'request': [b'>\x00', b'"\xde\x02'],
                          'responseAddress': 1848,
                          'subAddress': 0},
                         {'address': 1888,
                          'brand': 'ford',
                          'bus': 0,
                          'ecu': 'abs',
                          'fwVersion': b'\x01\x02\x00\x00\x01\x04\x00\x00\x00\x05\x03\x01\x02\x02\x01\x03\x01\x01\x01\x02\x01\x02\x01\x01'
                                       b'\x00\x00\x00',
                          'logging': True,
                          'obdMultiplexing': True,
                          'request': [b'>\x00', b'"\xde\x02'],
                          'responseAddress': 1896,
                          'subAddress': 0},
                         {'address': 1798,
                          'brand': 'ford',
                          'bus': 0,
                          'ecu': 'fwdCamera',
                          'fwVersion': b"\xff\xfc'\xc3\x88\x00\x00\x00\x00\x00",
                          'logging': True,
                          'obdMultiplexing': True,
                          'request': [b'>\x00', b'"\xde\x03'],
                          'responseAddress': 1806,
                          'subAddress': 0},
                         {'address': 1840,
                          'brand': 'mazda',
                          'bus': 0,
                          'ecu': 'eps',
                          'fwVersion': b'LX6C-14D003-AL\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
                          'logging': False,
                          'obdMultiplexing': True,
                          'request': [b'"\xf1\x88'],
                          'responseAddress': 1848,
                          'subAddress': 0},
                         {'address': 1888,
                          'brand': 'mazda',
                          'bus': 0,
                          'ecu': 'abs',
                          'fwVersion': b'LX6C-2D053-RF\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
                          'logging': False,
                          'obdMultiplexing': True,
                          'request': [b'"\xf1\x88'],
                          'responseAddress': 1896,
                          'subAddress': 0},
                         {'address': 1798,
                          'brand': 'mazda',
                          'bus': 0,
                          'ecu': 'fwdCamera',
                          'fwVersion': b'M1PT-14F397-AD\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
                          'logging': False,
                          'obdMultiplexing': True,
                          'request': [b'"\xf1\x88'],
                          'responseAddress': 1806,
                          'subAddress': 0},
                         {'address': 1892,
                          'brand': 'mazda',
                          'bus': 0,
                          'ecu': 'fwdRadar',
                          'fwVersion': b'LB5T-14D049-AB\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
                          'logging': False,
                          'obdMultiplexing': True,
                          'request': [b'"\xf1\x88'],
                          'responseAddress': 1900,
                          'subAddress': 0}],
```

### CODE REVIEW

**Comment by sshane** [comment]
@incognitojam does https://github.com/commaai/openpilot/pull/31124 fingerprint this

**Comment by incognitojam** [comment]
Yes
image

File: selfdrive/car/ford/fingerprints.py [hunk 1]
```diff
@@ -8,16 +8,19 @@
     (Ecu.eps, 0x730, None): [
       b'LX6C-14D003-AH\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
       b'LX6C-14D003-AK\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
+      b'LX6C-14D003-AL\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
```
Comment (sshane) [inline selfdrive/car/ford/fingerprints.py:11]:
```suggestion
      b'LX6C-14D003-AL\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
```

File: selfdrive/car/ford/fingerprints.py [hunk 2]
```diff
@@ -8,16 +8,19 @@
...
     ],
     (Ecu.abs, 0x760, None): [
       b'LX6C-2D053-RD\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
       b'LX6C-2D053-RE\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
+      b'LX6C-2D053-RF\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
```
Comment (sshane) [inline selfdrive/car/ford/fingerprints.py:16]:
```suggestion
      b'LX6C-2D053-RF\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
```

File: selfdrive/car/ford/fingerprints.py [hunk 3]
```diff
@@ -8,16 +8,19 @@
...
     ],
     (Ecu.fwdRadar, 0x764, None): [
       b'LB5T-14D049-AB\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
     ],
     (Ecu.fwdCamera, 0x706, None): [
       b'M1PT-14F397-AC\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
+      b'M1PT-14F397-AD\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
```
Comment (sshane) [inline selfdrive/car/ford/fingerprints.py:23]:
```suggestion
      b'M1PT-14F397-AD\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
```

**Comment by sshane** [comment]
| Dongle, platform, VIN                                                                                                                | Name from VIN 🆚 CarInfo                               | Segments                | Bad seg tags   | Good seg tags                                        |
|:-------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------|:------------------------|:---------------|:-----------------------------------------------------|
| [4cc0b0d181c79cb1](https://useradmin.comma.ai/?onebox=4cc0b0d181c79cb1)<br><sub>FORD BRONCO SPORT 1ST GEN<br>3FMCR9C66........</sub> | FORD Bronco Sport 2023 🆚<br>Ford Bronco Sport 2021-23 | 16 routes, 184 segments |                | - all lat active: 29<br>- no input all lat active: 5 |

**Pull request merged**